### DISCUSSION TITLE
-favicon flag

### DISCUSSION THREAD

**Discussion by c4ir0** [projectdiscovery/httpx#834] [Ideas]
### my talk abiut -favicon, flag I know it's simple to use just pass the domain name and it will calculate the hash and give U lovely result like :

`echo www.domain.com | httpx -favicon `
the function adds /favicon.ico suffix to the url you added then start to calculate the hash and that might be poor implementaion for some reasons :

**1 - I could pass full url [as I usually was doing before I notice ] like :**

![1](https://user-images.githubusercontent.com/108268587/198100126-a3f58a83-c4c2-4c2b-a3eb-fd8bd866af16.JPG)

the tool her just adds the suffix [/favicon.ico ]to the base URL and clculate the hash and the result here is totally wrong [as no valid url ends up with ../favicon.ico/favicon.ico ]

**2 - some may say it's easy U could just pass the domain name not full url, I'd say ur right this really might work [might !], see these websites real urls of the favicon.ico :**

https://static.domain.com/images/favicon.ico
https://domain.org/static/assets/favicon.ico
https://static.domain.com/gallery/pics/another-sub-path/.../favicon.ico

You will notice that the favicon may existed in nested path and doesn't existed in the / or maybe hosted in a subdomain "static.domain.tld" and just adding the suffix will get U another wrong result [in cases I faced alot ]

## 💡 Solution :

**Hint: just get the real favicon.ico url and then start to calculate the hash like :**

![2](https://user-images.githubusercontent.com/108268587/198100154-82e16668-1dee-4835-a26d-4b881c68cdbd.JPG)

### so my suggestion is : U pass the domain name to httpx --> httpx requests the domain then --> extracts the favicon url --> calculates the hash --> totally correct result 😺

+ **advanced_Enhancement_1: You may find the favicon exists as an endpoint not full url "`/static/assets/favicon.ico`" so the tool may check if the extraction doesn't have http:// at the begining of it, if so it adds the https://domain.com before the endpoint then --> calculation** 

+ **advanced_Enhancement_2 : the favicon may not exist at all in the domain U requested so the tool may have to get it passively through (e.g. https://archive.org/web/)**

**Comment by ehsandeep** [comment]
@c4ir0 this already fixed in https://github.com/projectdiscovery/httpx/releases/tag/v1.2.5

**Reply by c4ir0** [reply]
> @c4ir0 this already fixed in https://github.com/projectdiscovery/httpx/releases/tag/v1.2.5

What's the way to update ?

**Reply by ehsandeep** [reply]
https://github.com/projectdiscovery/httpx#installation-instructions, installing/updating step is the same.

**Reply by c4ir0** [reply]
> https://github.com/projectdiscovery/httpx#installation-instructions, installing/updating step is the same.

Thx UwU
